### DISCUSSION TITLE
NextCloud AIO Font Upload

### DISCUSSION THREAD

**Discussion by armouredking** [nextcloud/all-in-one#1859] [Questions] [answered]
So I had this setup on my previous instance by creating a volume and adding it. Saw that it was a feature with AIO in #870 #1032 but also saw there were issues from #1643.

I am also experiencing the same issues of #1643. I have the TTF from before, and I wanted to add them per the procedure for AIO but each upload regardless of file attempted results in an error `Font Format Not Supported`.

There is no entry made into NextCloud logs, or the logs shown in AIO page, that I can find happening when I attempt this, so unsure how to proceed with troubleshooting.

**Comment by szaimen** [comment]
> Font Format Not Supported

Yes, I can reproduce that on my test instance. However since nothing gets uploaded not even in the browser, I supspect a problem with the upload code by the richdocuments app. Feel free to open a new issue here: https://github.com/nextcloud/richdocuments/issues

**Reply by armouredking** [reply]
So, the problem is, I went ahead and built an instance from scratch for NextCloud (not using docker, just bare metal install) and the font upload works just fine. So it is likely not an issue with Nextcloud's rich documents, but with some setting or configuration setup of the AIO system. Unfortunately, without some log somewhere to give a heading, I don't know how to proceed. Nothing I can see even happens in the browser debug/network tabs.

**Reply by szaimen** [reply]
I am in the same position as you. I don't have any clue what goes wrong and I also don't see any obvious configuration problem. Also without any log whatsoever it is not possible to debug this.

Additionally, I know that some people were able to upload fonts on AIO so maybe it is only happening on specific browser with OS, so it is likely not a specific AIO problem...

**Comment by szaimen** [comment]
I think I might have found one of the problems and ir is indeed in the richsocuments app.

**Reply by szaimen** [reply]
Compare e.g. https://github.com/nextcloud/richdocuments/blob/c4c3c5fc6f510fe1656d5b0b52a3b9183201b61e/src/components/AdminSettings.vue#L418 with https://github.com/nextcloud/server/blob/37bb33c5799b834dfef3fb73936bd0e5a4773fb8/resources/config/mimetypemapping.dist.json#L192

**Comment by szaimen** [comment]
Okay, I checked and it seems like the api used is just broken on my windows device: https://developer.mozilla.org/en-US/docs/Web/API/File/type. It always returns unknown type if I try to upload a font. So definitely nothing AIO could fix.

**Reply by szaimen** [reply]
On Linux it seems to work: 
![image](https://user-images.githubusercontent.com/42591237/216109013-42502977-2048-4a5c-a446-ce79f7a0632a.png)

**Reply by szaimen** [reply]
Seems like this is the problem on windows:
![image](https://user-images.githubusercontent.com/42591237/216109425-251672ab-13c5-4c2b-8c78-fcdadd92f598.png)
vs
![image](https://user-images.githubusercontent.com/42591237/216109612-2c209f52-b9e8-4f9b-be7e-9d8b66cc21c8.png)
and 
![image](https://user-images.githubusercontent.com/42591237/216109708-43050c1d-8ce2-4b26-9b26-f196668542b8.png)

**Reply by armouredking** [reply]
Interesting that Windows would be blocking for you... I'm principly using PopOS Firefox 108.0.2. I did also try with Windows Server 2016 Firefox 109.0, Internet Explorer (not sure version matters here), and Edge 109.0.1518.70 though, to no avail. The upload does however work when Nextcloud/Collabora is installed bare metal.

**Reply by szaimen** [reply]
Maybe popos has the same issue? Can you check the mimetype mapping? Do you always get the `Font Format Not Supported` issue or a different and is the mimetype added there?

**Reply by szaimen** [reply]
Also as I said was the api in general not working in my case and after applying the fix it works. So definitely unrelated to aio.

**Reply by armouredking** [reply]
It's always Font Format Not Supported () when trying with AIO regardless. Don't see anything in Dev Tools for Firefox or Edge. Nor anything happening in any logs.

PopOS shows `cat /etc/mime.types | grep ttf` as `font/ttf     ttf`

Using bare metal Collabora (packages from their repo) and Nextcloud (zip from website, with initial app wizard install for Office) - ie no docker - works out-of-box for font uploads from both PopOS and Windows 2016.

**Reply by szaimen** [reply]
> It's always Font Format Not Supported ()

All right. This means the api does not work correctly. Can you on the same devices check what the upload picker returns with https://developer.mozilla.org/en-US/docs/Web/API/File/type

**Reply by armouredking** [reply]
Think a comment got deleted or moved, original reply:
>I don't have screenshots unfortunately. Part of my move to do the full kettle install was to test performance because I was having another odd issue with AIO or Docker NextCloud using a ZFS store for the NCDATA path. It would take several seconds for Files to show up using a Docker suite, whereas bare metal Files loads in at most 2 seconds. That is probably a me / Docker config issue though.

>I think the message was from Push, or maybe Nextcloud's service, because it's a light red banner on the upper right part of the webpage for a few seconds that says `Font Format Not Supported ()` and I would see the push message in the Network tab of F12 at the same time.

Reply to above:
The tool returns `arial.ttf: font/ttf`

Should also note, that I have been attempting the font upload with the file shared in #1643 to keep as much the same as possible. Testing on Windows with that file, or with one of my fonts does result in a response of `calibri.ttf: unknown` from the tool, so at least on the Windows side of things it's probably related to mime types. I'm not sure why PopOS gives good data but still had issues uploading fonts though.

**Reply by szaimen** [reply]
> Testing on Windows with that file, or with one of my fonts does result in a response of `calibri.ttf: unknown` from the tool, so at least on the Windows side of things it's probably related to mime types.

All right. Can you check if applying my steps below make it work on windows? (probably a restart is needed).

**Reply by armouredking** [reply]
It appears to work after making the regedits. Not sure how I feel about telling people to make regedits, but providing a registry key would also be kinda eh. Maybe an advanced separate section that has warnings plastered about making sure you backup before you touch.

**Reply by szaimen** [reply]
Good that it works. I already told the people in question that the API that they are using is unreliable, so they will look into possible replacements in the future.

**Comment by szaimen** [comment, marked as answer]
Nice: after adding a Content Type REG_SZ, it allows me to upload the fonts:
![image](https://user-images.githubusercontent.com/42591237/216111908-cc04c8d2-4475-4a63-b74e-4243d50a8465.png)
![image](https://user-images.githubusercontent.com/42591237/216111984-70ec81b2-7c96-4ed0-beed-7ccae592bb8f.png)
